### PR TITLE
Make FileVersionProvider repleacable

### LINKED ISSUE #6371: FileVersionProvider should be replaceable
Either though DI or an option so that devs can decide to use their own file eviction logic (timestamp for instance) or a faster hash.

### PR DESCRIPTION
**Pull request by pranavkm**
Fixes #6371

### CODE REVIEW

File: src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs [hunk 1]
```diff
@@ -228,9 +228,11 @@ internal static void AddRazorViewEngineServices(IServiceCollection services)
             // TagHelperComponents manager
             services.TryAddScoped<ITagHelperComponentManager, TagHelperComponentManager>();
 
-            // Consumed by the Cache tag helper to cache results across the lifetime of the application.
+            // Infrastructure for MVC TagHelpers
             services.TryAddSingleton<IMemoryCache, MemoryCache>();
             services.TryAddSingleton<TagHelperMemoryCacheProvider>();
+
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.Razor/DependencyInjection/MvcRazorMvcCoreBuilderExtensions.cs]: isn't the file version provider also part of the infrastructure?

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/ImageTagHelper.cs [hunk 2]
```diff
@@ -63,12 +61,14 @@ public class ImageTagHelper : UrlResolutionTagHelper
         public ImageTagHelper(
             IHostingEnvironment hostingEnvironment,
             TagHelperMemoryCacheProvider cacheProvider,
+            IFileVersionProvider fileVersionProvider,
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/ImageTagHelper.cs]: Just double checking, this ctor hasn't shipped yet so we can add this param?

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/ImageTagHelper.cs [hunk 3]
```diff
@@ -121,18 +123,15 @@ public override void Process(TagHelperContext context, TagHelperOutput output)
                 // not function properly.
                 Src = output.Attributes[SrcAttributeName].Value as string;
 
-                output.Attributes.SetAttribute(SrcAttributeName, _fileVersionProvider.AddFileVersionToPath(Src));
+                output.Attributes.SetAttribute(SrcAttributeName, FileVersionProvider.AddFileVersionToPath(ViewContext.HttpContext.Request.PathBase, Src));
```
Comment (NTaylorMullen) [inline src/Microsoft.AspNetCore.Mvc.TagHelpers/ImageTagHelper.cs]: If you're feeling inspired refactor the file version into a variable to make debugging easier. If not, good as is. Same true for other TagHelpers.

File: src/Microsoft.AspNetCore.Mvc.Razor/Infrastructure/IFileVersionProvider.cs [hunk 4]
```diff
@@ -0,0 +1,21 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using Microsoft.AspNetCore.Http;
+
+namespace Microsoft.AspNetCore.Mvc.Razor.Infrastructure
```
Comment (rynowak) [inline src/Microsoft.AspNetCore.Mvc.Razor/Infrastructure/IFileVersionProvider.cs]: Consider just `Mvc.Infrastructure` for this. It's not really tied to Razor in any way

File: src/Microsoft.AspNetCore.Mvc.TagHelpers/ImageTagHelper.cs (hunk 3 above)
Comment (rynowak) [reply src/Microsoft.AspNetCore.Mvc.TagHelpers/ImageTagHelper.cs]: meh

**Pull request closed without merging**